packages/grafana-data/src/events/types.ts

```diff
@@ -76,6 +72,13 @@ export interface EventBus {
    * Remove all event subscriptions
    */
   removeAllListeners(): void;
+
+  /**
+   * Returns a new bus scoped that knows where it exists in a heiarchy
+   *
+   * @internal -- This is included for internal use only should not be used directly
+   */
+  newScopedBus(key: string, localOnly?: boolean): EventBus;
```

Comment:
We currently want: `dashboard > panel`, but very very soon we will want:  `dashboard > panel > legend` etc.